### PR TITLE
Add auth0 login

### LINKED ISSUE #3: Implement Auth0 workflow
The application should be able to log in to Auth0 and acquire a JWT token to be passed over to other services.

Some guidelines: https://infosec.mozilla.org/guidelines/iam/openid_connect.html#authorization-code-grant

### PR DESCRIPTION
**Pull request by magopian**
Fixes #3

### CODE REVIEW

**Comment by magopian** [comment]
I guess this is now ready to review, @andymikulski maybe?

File: public/index.html [hunk 1]
```diff
@@ -20,6 +20,7 @@
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
     <title>Delivery Console</title>
+    <script src="https://cdn.auth0.com/js/auth0/8.9/auth0.min.js"></script>
```
Comment (andymikulski) [inline public/index.html]: Is there a reason to pull this from a CDN over Yarn or NPM?

File: src/console/auth0.test.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  initWebAuth,
+  isAuthenticated,
+  login,
+  logout,
+  setSession,
+  webAuthHandler,
+} from './auth0';
+
+// Mock the localStorage API.
+global.localStorage = (function() {
```
Comment (andymikulski) [inline src/console/auth0.test.js]: We also mock `localStorage` in `setupTests.js`, is this necessary here?

File: public/index.html (hunk 1 above)
Comment (magopian) [reply public/index.html]: nope, absolutely none :D

**Pull request merged**